vortex-datafusion/src/persistent/source.rs

```diff
@@ -135,7 +137,10 @@ impl FileSource for VortexSource {
     }
 
     fn metrics(&self) -> &ExecutionPlanMetricsSet {
-        self.metrics.report_to_datafusion()
+        // We currently report empty metrics back to DataFusion since it can be expensive
+        // to flush our own metrics into the DataFusion format. We should revisit this in future
+        // if we want tighter integration.
```

Comment:
I think part of the problem is that this is called in `FileScanConfig::open` from datafusion, so doing any mutation to metrics here will only accumulate on each scan